### ISSUE TITLE
How to build from source on MSYS2 MINGW64?

### ISSUE THREAD

**Issue by iahung** [vkohaupt/vokoscreenNG#266]

**Comment by vkohaupt** [comment]
I don't know any MSYS2, what is that good for? I don't want to work on something I don't need. A little more information from you would be nice.

**Comment by iahung** [comment]
> I don't know any MSYS2, what is that good for? I don't want to work on something I don't need. A little more information from you would be nice.

MSYS2 is a Linux like build environment for Windows, but unlike Cygwin which is a POSIX compatibility layer, MSYS2 produces native Win32 programs that doesn't need any POSIX compatibility layer to run (assuming you are not using the MINGW64 Shell).

**Comment by vkohaupt** [comment]
MSYS2 is not listed here.
https://doc.qt.io/qtcreator/creator-tool-chains.html

Does this answer your question?

**Comment by iahung** [comment]
> MSYS2 is not listed here. https://doc.qt.io/qtcreator/creator-tool-chains.html
> 
> Does this answer your question?

Nope. If you can please do a bit more research before posting to show some respects for the people you are talking with.

Well, MSYS2 MINGW64 is basically... MinGW, the MinGW-w64 compiler packaged together with a Linux-like build environment. Qt has always supported MinGW. As you can see, they always provide two version: one is for MSVC, one is for MinGW. I remember Qt even offers MinGW packaged as part of the MinGW version.

If you need Qt Creator, you could install it directly on MSYS2 MINGW64 via the package:

https://packages.msys2.org/package/mingw-w64-x86_64-qt-creator

This Qt Creator is Qt6 based. Most of the time, to build Qt project you only need `qmake`. `qmake` is packaged as part of Qt.

For Qt5: install `mingw-w64-x86_64-qt5` package, `qmake` binary is `qmake.exe`

For Qt6: install `mingw-w64-x86_64-qt6` package, `qmake` binary is `qmake6.exe`

For other projects, `qmake project.pro` then `make -j8` and done. You are not even need to launch Qt Creator. But this doesn't work straight forward with your project. I suspect the problem is with resources and translations so you need to update your `preparing.sh` for MSYS2 MINGW64.
